### ISSUE TITLE
Screenshot of Splunk Dashboard

### ISSUE THREAD

**Issue by WhiteBoardDev** [rheobot-team/rheobot#5]
I would like to have a command where I can see a screenshot of a splunk dashboard.

1. somehow get a list of available dashboards
2. choose one to get a screen shot of
3. have authentication and authorization handled for me

**Comment by jonatan-ivanov** [comment]
How about screenshot of any site? :)
Login will be an interesting question.

Also: https://github.com/GoogleChrome/puppeteer

**Comment by WhiteBoardDev** [comment]
Good point. I was trying to be more specific so we had a particular use case to go after. However, I do agree that if we start by making a screen shot of any site we can make this feature config driven for any site.

I would like to make sure it can handle auth somehow. Would we have to support cookies for sites that have a dedicated login page? would one of these libraries be able to follow 302 redirects to login pages and then bounce back?

**Comment by srt4** [comment]
For cases where we would have to login, I think there would need to be site-specific implementations, or at least configurations (making the suggestion instead, "screenshot of any [non-authenticated] site"). 

For example, at the very least, we'd have to configure the element ID for the username and password inputs.

**Comment by WhiteBoardDev** [comment]
I think we should bake in some common auth methods to make it more user friendly. I imagine a config something like

```
{
"site" : "myurl",
"authentication" : {
   "basic": {
   "username":"user",
"password":"pass"
}
}

}
```

**Comment by srt4** [comment]
I'm going to look at adding some other issues to this project, because I'm worried about starting off with this one. It feels like it may be massive in scope, and may not be accomplish-able in Hacktoberfest time. (I'd be happy to be convinced otherwise, though) 

Some areas of complexity I'm seeing:
 
(1) With Splunk in particular, how do we determine when to take the snapshot? We can't use, for example, `$(document).ready()` style event handling - since the page finishes "loading" from the browser perspective even while the query is still running. We'd have to look in the Splunk JS for some sort of `onQueryResultsLoaded` event. This would apply to every arbitrary website we may want to screenshot. 

(2) Do we screenshot the whole page? (Might be ugly/not user-friendly). If not, do we configure a viewport? (e.g. capture a 900px by 500px box?) Configured per site? 

(3) Authentication can get messy but I think it's doable. 

(4) Overall, due to its dependency on headless Chrome and its possibly-needing access to resources on a different network, this feels like something we'd want to bundle and ship as a Docker container, exposing an endpoint which a Hubot component could call. 

Thoughts?

**Comment by jonatan-ivanov** [comment]
(1) I have no idea. :)
(2) Yepp, viewport is a kind of a must here.
(3) Agree.
(4) That is the really messy thing, installing headless chrome can be a headache on some linux distros.

Since this seems big, we can also decide dropping this and implement post-hacktoberfest or we all can work on this to get it done.
